quipucords/scanner/network/inspect_callback.py

```diff
@@ -1,239 +1,298 @@
+            self._ansible_facts[host][fact_name] = fact_value
 
     def task_on_failed(self, event_dict):
-        """Print a json representation of the event_data on failed."""
-        event_data = event_dict.get("event_data")
-        host = event_data.get("host", UNKNOWN)
+        """Handle failure events."""
+        # example event
+        # {
+        #   "uuid": "5a29fb37-da65-4d2e-92f6-826b924ccfbf",
+        #   "counter": 125,
+        #   "stdout": "fatal: [127.0.0.1]: FAILED! => <...>",
+        #   "start_line": 309,
+        #   "end_line": 321,
+        #   "runner_ident": "13ff256b-b793-47d7-b26f-9028f05d9e09",
+        #   "event": "runner_on_failed",
+        #   "pid": 74971,
+        #   "created": "2024-02-02T11:49:38.377767",
+        #   "parent_uuid": "6c946626-0f07-7c1b-37f1-000000000035",
+        #   "event_data": {
+        #     "playbook": "<...>",
+        #     "playbook_uuid": "a0ff6f63-c1b8-46fe-ab9c-092a50cffb28",
+        #     "play": "group_0",
+        #     "play_uuid": "6c946626-0f07-7c1b-37f1-000000000002",
+        #     "play_pattern": " group_0 ",
+        #     "task": "gather internal_have_java_cmd",
+        #     "task_uuid": "6c946626-0f07-7c1b-37f1-000000000035",
+        #     "task_action": "raw",
+        #     "resolved_action": "ansible.builtin.raw",
+        #     "task_args": "",
+        #     "task_path": "<...>",
+        #     "role": "check_dependencies",
+        #     "host": "127.0.0.1",
+        #     "remote_addr": "127.0.0.1",
+        #     "res": {
+        #       "rc": 1,
+        #       "stdout": "",
+        #       "stdout_lines": [],
+        #       "stderr": "Shared connection to 127.0.0.1 closed.\r\n",
+        #       "stderr_lines": [
+        #         "Shared connection to 127.0.0.1 closed."
+        #       ],
+        #       "changed": true,
+        #       "msg": "non-zero return code",
+        #       "_ansible_no_log": null
+        #     },
+        #     "start": "2024-02-02T11:49:38.299511",
+        #     "end": "2024-02-02T11:49:38.377629",
+        #     "duration": 0.078118,
+        #     "ignore_errors": true,
+        #     "event_loop": null,
+        #     "uuid": "5a29fb37-da65-4d2e-92f6-826b924ccfbf"
+        #   }
+        # }
+        event_data = event_dict.get("event_data", {})
+        host = event_data.get("host", UNKNOWN_HOST)
         result = event_data.get("res")
+        task = event_data.get("task")
+        role = event_data.get("role")
         if result.get("rc") == TIMEOUT_RC:
-            logger.warning("Task %s timed out", event_data["task"])
+            logger.error("[host=%s] Task %s timed out", host, event_data.get("task"))
 
         # Only log an error when ignore errors is false
         if event_dict.get("ignore_errors", False):
             err_reason = result.get("msg", event_data)
+            # is it unexpected, tho? we have so many commands that will recurrently fail
```

Comment:
Agreed… maybe the message and log level should be `warning` instead. Maybe? I'm not 100% sure either. 🤔

Reply:
changed it to warning